pandas/tests/io/formats/test_to_csv.py

```diff
@@ -223,3 +227,34 @@ def test_to_csv_multi_index(self):
 
         exp = "foo\nbar\n1\n"
         assert df.to_csv(index=False) == exp
+
+    @pytest.mark.parametrize('filename,compression,compressor', [
+        ("test.gz", "gzip", gzip.GzipFile),
+        ("test.gz", "infer", gzip.GzipFile),
+        ("test.bz2", "bz2", bz2.BZ2File),
+        ("test.bz2", "infer", bz2.BZ2File)
+    ])
+    def test_to_csv_compression(self, filename, compression, compressor):
+        # see GH issue 15008
+
+        df = DataFrame([1])
+        exp = "".join((",0", os.linesep, "0,1", os.linesep)).encode("ascii")
+
+        with tm.ensure_clean(filename) as path:
+            df.to_csv(path, compression=compression)
+            with compressor(path, "rb") as f:
+                assert f.read() == exp
+
+    @pytest.mark.parametrize('compression', ["xz", "infer"])
+    def test_to_csv_compression_lzma(self, compression):
+        # see GH issue 15008
+
+        lzma = tm._skip_if_no_lzma()
+
+        df = DataFrame([1])
+        exp = "".join((",0", os.linesep, "0,1", os.linesep)).encode("ascii")
+
+        with tm.ensure_clean("test.xz") as path:
```

Comment:
add an additional comparison that reads back these files (you can simply use infer), then ``assert_frame_equal`` to the original.